### DISCUSSION TITLE
"Extended VSCode" API

### DISCUSSION THREAD

**Discussion by mauricioszabo** [pulsar-edit/.github#80] [Core Application & Packages]
Considering that multiple Atom packages are being archived right now, I was thinking on something to re-use packages from VSCode: an "extended VSCode" API.

Meaning, basically: we somehow support VSCode API, translating the calls to the old Atom API, but add some information (on the VSCode API) that says we have an "extended" version of it. 

An example could be: VSCode offers a "decorations" API that supports showing stuff above or below some line. I believe VSCode only supports Markdown, but if we add, on this API, an info like "we do support HTML", somebody could write a package that would branch to our "extended HTML" support or fallback to VSCode - effectively publishing a package that works on both editors, but it's better on Pulsar. 

This can re-spark interest on Pulsar, and somehow mitigate the problems we have with breaking changes on older packages if we decide to keep pushing for more recent Electron versions...

**Comment by Daeraxa** [comment]
Whilst in theory I'm loathe to have to capitulate to VSCode I think it does make sense that we support it as that is where all the development has gone, Atom packages have been largely forgotten in favour of VSCode extensions.
